### DISCUSSION TITLE
catch flow cancel error

### DISCUSSION THREAD

**Discussion by DMS40** [mobxjs/mobx#2789] [Q&A] [answered]
Hello, I have some question about mobx's [flow](https://mobx.js.org/actions.html)

#### Case
I use react, and call mobx flow method from component's useEffect hook.
When useEffect's clear function is called, I want to call flow's `cancel()` and I did it.

it's what I want exactly! 

But I met this message
![스크린샷 2021-02-05 오후 7 22 46](https://user-images.githubusercontent.com/78593468/107021499-8916a100-67e7-11eb-9eb5-16e731fc4a1a.png)

That makes my integrate test fail and I don't want to show this message from chrome developement console.

#### Question
Can I catch this error? I tried but failed. or Is it unexpected case?

Thank you for your time

**Comment by mweststrate** [comment]
You should be able to catch it like you catch any promise error

**Reply by DMS40** [reply]
@mweststrate I tried but can't catch it 😭

This is what I tested, Like mobx example

```js
class Store {
    state = 'pending'

    constructor() {
      makeAutoObservable(this, {
        fetchProjects: flow,
      });
    }

    * fetchProjects() {
      this.state = 'pending';

      try {
        const result = yield new Promise((res, rej) => {
          setTimeout(() => {
            res('good');
          }, 3000);
        });
        // not fired by cancel()
        console.log(result);

        this.state = 'done';

      } catch (error) {
        //here ? 
        this.state = 'error';
      }
    }
}

//Will be in React App
(() => {
  const store = new Store();

  try{
    const some = flowResult(store.fetchProjects());

    // will fired useEffect's clear function
    some.cancel();
  }catch(e) {
    //or here ? 
    console.log(e);
  }
})();
```

**Reply by mweststrate** [reply]
hmm that should work I think, best file a bug report including a small reproduction in codesandbox.

**Reply by DMS40** [reply]
@mweststrate How can I report ? And this one can be example? 
https://codesandbox.io/s/magical-hugle-qc0d1?file=/src/App.js

Thank you for your kindness!

**Comment by mweststrate** [comment]
https://github.com/mobxjs/mobx/issues/new/choose

**Comment by iChenLei** [comment, marked as answer]
solved by https://github.com/mobxjs/mobx/issues/2790#issuecomment-774287633
